test-harness

```diff
@@ -211,11 +216,19 @@ def main():
         # enqueues execution of
         # evaluate(name, code, config_file, **opts)
         # in the thread pool
-        tp.submit(evaluate, name, code, config_file, **opts)
+        jobs.append(tp.submit(evaluate, name, code, config_file, **opts))
 
     # Wait for all jobs to finish
     tp.shutdown(wait=True)
 
+    # Check that none of the jobs raised an exception:
+    exceptions = [ j.exception() for j in jobs ]
+    if any(exceptions):
+      raise SystemExit(("Error: One of the test jobs failed to execute, producing"
```

Comment:
I think you should print the name of the job.

Reply:
good point, done